### PR TITLE
DOC: update the pandas.DataFrame.all docstring

### PR DESCRIPTION
**Pull request by mlaforet**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [x] PR title is "DOC: update the <your-function-or-method> docstring"
- [x] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [x] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
################################################################################
####################### Docstring (pandas.DataFrame.all) #######################
################################################################################

Return whether all elements are True over requested axis.

Returns True if all elements along a dataframe
axis are non-zero, not-empty or not-False.
Also note that a series consisting of different data
types returns the first occurence of the non-zero, not-empty
or not-False element.

Parameters
----------
axis : {index (0), columns (1)}
skipna : boolean, default True
    Exclude NA/null values. If an entire row/column is NA, the result
    will be NA.
level : int or level name, default None
    If the axis is a MultiIndex (hierarchical), count along a
    particular level, collapsing into a Series.
bool_only : boolean, default None
    Include only boolean columns. If None, will attempt to use everything,
    then use only boolean data. Not implemented for Series.
**kwargs : any, default None
    Additional keywords have no fect but might be accepted for
    compatibility with numpy.

Returns
-------
all : Series or DataFrame (if level specified)

Examples
--------
First create a pandas dataframe::
    >>> d = {'col1': [True, True], 'col2': [True, False]}
    >>> df = pd.DataFrame(data=d)
    >>> df
       col1   col2
    0  True   True
    1  True  False

Default behaviour checks if column-wise values all return True
    >>> df.all()
    col1     True
    col2    False
    dtype: bool

Adding axis=1 will check row-wise values
    >>> df.all(axis=1)
    0     True
    1    False
    dtype: bool

See also
--------
pandas.DataFrame.any : Checks if one (or more) items return True

################################################################################
################################## Validation ##################################
################################################################################

Errors found:
	Errors in parameters section
		Parameters {'kwargs'} not documented
		Unknown parameters {'**kwargs'}
		Parameter "axis" has no description

```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.
The axis description is a bug in the string interpollation of %(axis_descr)s
The **kwargs error is a bug in the parameter evaluation

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -7576,11 +7523,10 @@ def _add_numeric_operations(cls):
         cls.any = _make_logical_function(
             cls, 'any', name, name2, axis_descr,
             'Return whether any element is True over requested axis',
-            nanops.nanany)
+            nanops.nanany, '', '')
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: Maybe remove these so they don't conflict with the other PR.

File: pandas/core/generic.py [hunk 2]
```diff
@@ -7837,25 +7783,69 @@ def _doc_parms(cls):
...
+    will be NA.
 level : int or level name, default None
     If the axis is a MultiIndex (hierarchical), count along a
-    particular level, collapsing into a %(name1)s
+    particular level, collapsing into a %(name1)s.
 bool_only : boolean, default None
     Include only boolean columns. If None, will attempt to use everything,
     then use only boolean data. Not implemented for Series.
+**kwargs : any, default None
+    Additional keywords have no fect but might be accepted for
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: fect -> affect.

File: pandas/core/generic.py [hunk 3]
```diff
@@ -7837,25 +7783,69 @@ def _doc_parms(cls):
...
+
+%(examples)s
+%(see_also)s"""
+
+_all_doc = """\
+Return whether all elements are True over requested axis.
+
+Returns True if all elements along a dataframe
+axis are non-zero, not-empty or not-False.
+Also note that a series consisting of different data
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: You can remove this second sentence. THe fact that this happens is a bug, so we don't want to document it.

File: pandas/core/generic.py [hunk 4]
```diff
@@ -7837,25 +7783,69 @@ def _doc_parms(cls):
...
+types returns the first occurence of the non-zero, not-empty
+or not-False element."""
+
+_all_examples = """\
+Examples
+--------
+First create a pandas dataframe::
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: No dobule colon.

File: pandas/core/generic.py [hunk 5]
```diff
@@ -7837,25 +7783,69 @@ def _doc_parms(cls):
...
+    >>> d = {'col1': [True, True], 'col2': [True, False]}
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: No indent.

**Review by TomAugspurger — COMMENTED** [review]
Seems like some unrelated changes slipped in here. Could you

1. Fetch from upstream
2. merge upstream/master
3. Make sure that the only changes are contained to the docstring for `all`

Otherwise we'll get a bunch of merge conflicts later on.

**Comment by jorisvandenbossche** [comment]
Be sure to also check the comments on the `DataFrame.any` PR https://github.com/pandas-dev/pandas/pull/20217 as it is a very related method, so both docstring are a bit consistent

File: pandas/core/generic.py (hunk 1 above)
Comment (mlafore3) [reply pandas/core/generic.py]: @TomAugspurger If I do this an error 
```python ../scripts/validate_docstrings.py pandas.DataFrame.all
Traceback (most recent call last):
  File "../scripts/validate_docstrings.py", line 37, in <module>
    import pandas
  File "/Users/marconlaforet/pandas-dev/pandas/__init__.py", line 42, in <module>
    from pandas.core.api import *
  File "/Users/marconlaforet/pandas-dev/pandas/core/api.py", line 10, in <module>
    from pandas.core.groupby import Grouper
  File "/Users/marconlaforet/pandas-dev/pandas/core/groupby.py", line 48, in <module>
    from pandas.core.frame import DataFrame
  File "/Users/marconlaforet/pandas-dev/pandas/core/frame.py", line 75, in <module>
    from pandas.core.series import Series
  File "/Users/marconlaforet/pandas-dev/pandas/core/series.py", line 3202, in <module>
    Series._add_numeric_operations()
  File "/Users/marconlaforet/pandas-dev/pandas/core/generic.py", line 7530, in _add_numeric_operations
    nanops.nanany)
TypeError: _make_logical_function() missing 2 required positional arguments: 'examples' and 'see_also'```

Comment (jorisvandenbossche) [reply pandas/core/generic.py]: Yes, you can keep this. Just depending on which one of the two PRs is merged first, you might need to solve the conflict on the other. But that should be no problem

**Comment by mlafore3** [comment]
I have made the requested changes and added docs/examples for series.all(). This PR is ready for review again. Let me know if there is anything else.

File: pandas/core/generic.py [hunk 6]
```diff
@@ -7841,25 +7840,77 @@ def _doc_parms(cls):
...
+0     True
+1    False
+dtype: bool
+"""
+
+_all_see_also = """\
+See also
+--------
+pandas.DataFrame.any : Return if one (or more) elements are True
+"""
```
Comment (jreback) [inline pandas/core/generic.py]: add Series.all

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/generic.py (hunk 6 above)
Comment (mlafore3) [reply pandas/core/generic.py]: @jreback its been added

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**